File: analysis/model/src/test/java/de/maibornwolff/codecharta/model/TreeCreator.java
package de.maibornwolff.codecharta.model;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class TreeCreator {
    public static Tree createTree() {
        return createTree(Path.TRIVIAL, null);
    }

    public static Tree createTree(final Path pathToInnerTree, final Tree innerTree) {
        return new Tree() {
            @Override
            public List<? extends Tree> getChildren() {
                return innerTree == null ? Collections.emptyList() : Arrays.asList(innerTree);
            }

            @Override
            public Path getPathOfChild(Tree child) {
                return pathToInnerTree;
            }

            @Override
            public String toString() {
                return pathToInnerTree + " -> " + innerTree;
            }
        };
    }
}

File: analysis/model/src/test/kotlin/de/maibornwolff/codecharta/model/TreeCreator.kt
package de.maibornwolff.codecharta.model

object TreeCreator {
    @JvmOverloads
    fun createTree(pathToInnerTree: Path = Path.TRIVIAL, innerTree: Tree<MutableNode>? = null): Tree<MutableNode> =
        object : Tree<MutableNode>() {
            override val children: Set<Tree<MutableNode>>
                get() = if (innerTree == null) emptySet() else mutableSetOf(innerTree)

            override fun getPathOfChild(child: Tree<MutableNode>): Path = pathToInnerTree

            override fun toString(): String = pathToInnerTree.toString() + " -> " + innerTree

            override fun insertAt(path: Path, node: MutableNode): Unit = throw NotImplementedError()
        }
}
